### ISSUE TITLE
add label alignement to tick_params

### ISSUE THREAD

**Issue by story645** [matplotlib/matplotlib#20644]
tick_params supports a subset of labeling parameters:
* labelcolor
* labelsize
* label{bottom, top, left, right}
* labelrotation

Motivated by wanting to swap out `plt.setp` and use `tick_params`, and finding the tick_params list kind of arbitrary, I think if labels can be modified via tick_params than a larger set of label parameters should be supported. This would mean adding:
* labelhorizantalingment
* labelverticalalignment
* labelfontproperties
* labelrotation_mode
* probably some pass through down to the text objects

**Comment by story645** [comment]
I totally agree w/ you on an automagical angle alignment as that's probably a common use case.

**Comment by story645** [comment]
Labeling good first issue because it seems to have support from at least two devs and is some what independent, but medium difficulty because it's kind of involved:
1) adding more dispatch in https://github.com/matplotlib/matplotlib/blob/v3.5.2/lib/matplotlib/axes/_base.py#L3358-L3435 and 
2) modifying https://github.com/matplotlib/matplotlib/blob/v3.5.2/lib/matplotlib/axis.py#L850-L887 to support the new kwargs
3) modify some methods on the tick class to support the new keywords https://github.com/matplotlib/matplotlib/blob/4f5cacfde9024076d9ab0cb6ad1c9a7cf352d5f9/lib/matplotlib/axis.py#L35

**Comment by QuLogic** [comment]
> (which I think means image tests?)

I think it can check `[label.get_[vh]a() for label in get_ticklabels()]`? (or via some similar new getter API if it gets added.)

**Comment by joooeey** [comment]
> (You could still use `setp()` if you need full control over all tick properties).

I've heard the `setp` excuse a lot all over this issue tracker and Stackoverflow in many contexts but it has a major flaw: For those of us working with interactive plots, we have to somehow make sure that `setp` is called after re-drawing when the ticks change.

Can anyone enlighten me what the reason is that we don't provide a mechanism to pass all keyword arguments to the ticks and labels through the various `.tick_params` and `.set_tick_params` methods? Is this just not implemented yet or are there major obstacles?

**Comment by timhoffm** [comment]
This ist mostly "just not implemented". However I except that there can be some corner cases (e.g. the alignment discussion above), where we might want to deviate from simply passing through.

**Comment by evelyn-lo** [comment]
Hi @timhoffm ! I'm Evelyn, and am interested in helping resolve this issue. Can me and my team work on this?

**Comment by timhoffm** [comment]
Hi @evelyn-lo thanks for your interest. I'm still unclear whether we should support the alignment. If so, it would have the effect as described [here](https://github.com/matplotlib/matplotlib/issues/13774#issuecomment-479503856), which should explicitly be mentioned as it may not be the obvious behavior.

**Comment by rcomer** [comment]
The use case that springs to mind for me is the third example here, where we add a bunch of spaces to move the month labels to the right.  Ideally we could set the horizontal_alignment to left instead

https://matplotlib.org/stable/gallery/ticks/multilevel_ticks.html
```python
# note the extra spaces in the label to align the month label inside the month.
# Note that this could have been done by changing ``bymonthday`` above as well:
sec.xaxis.set_major_formatter(mdates.DateFormatter('  %b'))
```

Admittedly this is a pretty specialised case.

**Comment by rcomer** [comment]
We should remove the GFI label while we decide if we still want this.

**Comment by timhoffm** [comment]
> The use case that springs to mind for me is the third example here, where we add a bunch of spaces to move the month labels to the right. Ideally we could set the horizontal_alignment to left instead

I don't think this would even help here. The day tick labels are center-aligned. The spacing emulates left-alignment to te day tick labels. but a possible `tick_params(horizontalalignment='left')` would left-align to the tick not to the day tick label. Alternatively left aligning bot the day label and the month label wrt. to the tick is possibly also not desirable here.

---

repeating https://github.com/matplotlib/matplotlib/pull/30009#issuecomment-2930483689:

> given, https://github.com/matplotlib/matplotlib/issues/20644#issuecomment-2528813561 and https://github.com/matplotlib/matplotlib/issues/20644#issuecomment-2819650076 i tend towards not implementing label alignment. Are there compelling use cases that still need to set this? If so, is the per-text alignment logic the right one or would one rather need a global function similar to `align_titels` \ `align_labels`?
